### PR TITLE
impl SizeOf for some basic `std::net` types

### PR DESCRIPTION
**Pull request by little-dude**

### CODE REVIEW

**Comment by little-dude** [comment]
I'm not sure whether you're open to PRs to support more third-party crates. If so I have [a branch](https://github.com/Kixiron/size-of/compare/master...little-dude:size-of:ipnetwork) that adds support for the [`ipnetwork`](https://crates.io/crates/ipnetwork) crate.

File: src/tests/mod.rs [hunk 1]
```diff
@@ -234,4 +238,24 @@ mod std {
...
+        assert_eq!(ip.size_of(), TotalSize::new(17, 0, 0, 0));
+        let addrv4 = SocketAddrV4::new(ipv4, 8080);
+        assert_eq!(addrv4.size_of(), TotalSize::new(6, 0, 0, 0));
+        let addrv6 = SocketAddrV6::new(ipv6, 1, 2, 3);
+        assert_eq!(addrv6.size_of(), TotalSize::new(28, 0, 0, 0));
+        let mut addr = SocketAddr::V4(addrv4);
+        assert_eq!(addr.size_of(), TotalSize::new(32, 0, 0, 0));
+        addr = SocketAddr::V6(addrv6);
+        assert_eq!(addr.size_of(), TotalSize::new(32, 0, 0, 0));
+    }
```
Comment (Kixiron) [inline src/tests/mod.rs]:
```suggestion
    #[test]
    fn socket_addresses() {
        let ipv4 = Ipv4Addr::new(127, 0, 0, 1);
        assert_eq!(ipv4.size_of(), TotalSize::total(4));

        let ipv6 = Ipv6Addr::from_str("::1").unwrap();
        assert_eq!(ipv6.size_of(), TotalSize::total(16));

        let mut ip = IpAddr::V4(ipv4);
        assert_eq!(ip.size_of(), TotalSize::total(17));

        ip = IpAddr::V6(ipv6);
        assert_eq!(ip.size_of(), TotalSize::total(17));

        let addrv4 = SocketAddrV4::new(ipv4, 8080);
        assert_eq!(addrv4.size_of(), TotalSize::total(6));

        let addrv6 = SocketAddrV6::new(ipv6, 1, 2, 3);
        assert_eq!(addrv6.size_of(), TotalSize::total(28));

        let mut addr = SocketAddr::V4(addrv4);
        assert_eq!(addr.size_of(), TotalSize::total(32));

        addr = SocketAddr::V6(addrv6);
        assert_eq!(addr.size_of(), TotalSize::total(32));
    }
```

**Review by Kixiron — CHANGES_REQUESTED** [review]

File: src/tests/mod.rs (hunk 1 above)
Comment (little-dude) [reply src/tests/mod.rs]: It does look better :) Applied

**Comment by little-dude** [comment]
@Kixiron would you mind taking another look? I addressed you comment.

**Review by Kixiron — APPROVED** [review]
Awesome, thanks!

**Comment by Kixiron** [comment]
And re `ipnetwork`, I'd gladly take a PR

**Comment by Kixiron** [comment]
And re `ipnetwork`, I'd gladly take a PR

**Pull request merged**